crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -389,18 +389,23 @@ def check_repo_tag
     end
 
     def check_key_file
-      expected = @config.fetch("repomd", {})["key_md5"]
+      expected = @config.fetch("repomd", {})["fingerprint"]
       return true if expected.blank?
 
       key_path = repodata_path.join("repomd.xml.key")
       key_path = repodata_media_path.join("repomd.xml.key") unless key_path.file?
 
       if key_path.file?
-        md5 = Digest::MD5.hexdigest(key_path.read)
+        fingerprint = `LC_ALL=C gpg --with-fingerprint #{key_path}`.split(/\r?\n/)
+        fingerprint.keep_if { |d| d =~ /fingerprint/ }
+        return false if fingerprint.empty?
+        fingerprint = fingerprint[0].split("=")
+        return false if fingerprint.length != 2
+        fingerprint = fingerprint[1].split.join(" ")
```

Comment:
I'm not sure but this block looks like it is possible to do with the ruby array map magic.